### DISCUSSION TITLE
Any plans for Integration with SvelteKit?

### DISCUSSION THREAD

**Discussion by Blankeos** [Thinkmill/keystatic#766] [Q&A] [answered]
My main framework right now is SvelteKit and absolutely would love to develop with keystatic on it. How complex is it to implement keystatic for another framework? Do you guys perhaps have a guide for integrating keystatic with just `@keystatic/core`?  Would be happy to hack around with it even if it takes time.

Also, is SvelteKit support also anywhere near the roadmap for this project?

**Comment by simonswiss** [comment, marked as answer]
Hey!

Keystatic is built with React and we have no concrete plans to build a version in a different framework.

However, you should be able to run Keystatic in a subdirectory in your SvelteKit project and run it as a separate Next.js app on a different port.

You can make it store the content at the root level of your project (where the SvelteKit app lives) and then read the files from there.

The other thing that will be a bit complicated is to render content from the document field. Keystatic provides the `DocumentRenderer` written in React — and you could take inspiration to write a similar renderer in Svelte.

The rest should work pretty well — if you're happy with that dual-nested-app setup.

Hope it helps!

**Reply by simonswiss** [reply]
Update: I made a [quick demo repo of Keystatic + SvelteKit](https://github.com/simonswiss/keystatic-sveltekit).

I haven't used SvelteKit before, so take the implementation with a grain of salt — but I've setup a SvelteKit project which hosts Keystatic as a "nested" Next.js App sitting in the /keystatic directory (and runs with a separate dev server).

The hard limitation out of the box is when you want to render the output of the document field, and cannot use the React DocumentRenderer.

I've JSON.stringify-ed the content instead for individual posts 😅

Keystatic itself is not deployed, just a pre-rendered site with the Keystatic content. You would deploy Keystatic as a separate deployment (Next.js app).

https://keystatic-sveltekit-simonswiss.vercel.app

**Reply by Blankeos** [reply]
Awesome! Thanks for this @simonswiss. I appreciate the demo!

**Comment by Greenheart** [comment]
Hi! Here's how I would integrate Keystatic with SvelteKit :)

I've just published an MIT-licensed integration of Keystatic CMS with SvelteKit and think many people will find it useful. https://github.com/Greenheart/keystatic-sveltekit

I also wrote a blog post explaining why this integration is useful, and how it was developed. This is a technical deep-dive into the development process that gives readers a better understanding of how SvelteKit, Vite and Keystatic work together. https://samuelplumppu.se/blog/keystatic-sveltekit-markdoc

I've been thinking about this in several projects, and now took the time to create a proper integration. I might open a PR for this to the main keystatic repo too if there's interest :blush:

**Reply by Greenheart** [reply]
I made significant updates and published `keystatic-sveltekit@1.0.0` to npm: https://www.npmjs.com/package/keystatic-sveltekit

This is now used in several production projects. Welcome to try it out and give feedback, submit issues and contribute PRs! :)

Once development of `@keystatic/core` picks up again, I'd be happy to deprecate the package and instead making it an official integration like for example `@keystatic/sveltekit`.
